backend/inspirehep/pidstore/cli.py

```diff
@@ -138,3 +138,50 @@ def mint_bais(yes_i_know, create):
             "There are still external BAIs left in DB. Probably they are not properly assigned to authors."
             " Please fix them and re-run command!"
         )
+
+
+@inspire_pidstore.command("fast-mint-new-bais")
+@click.option("--yes-i-know", is_flag=True)
+@with_appcontext
+def fast_mint_bais(yes_i_know):
+    """Mint BAIs"""
+    if not current_app.config.get("FEATURE_FLAG_ENABLE_BAI_PROVIDER"):
+        click.echo(
+            "Cannot mint BAIs without FEATURE_FLAG_ENABLE_BAI_PROVIDER set!", err=True
+        )
+        return
+    if current_app.config.get("FEATURE_FLAG_ENABLE_AUTHOR_DISAMBIGUATION"):
+        click.echo(
+            "Cannot mint BAIs with FEATURE_FLAG_ENABLE_AUTHOR_DISAMBIGUATION set!",
+            err=True,
+        )
+    if not yes_i_know:
+        click.confirm(
+            "Do you want to create new BAIs for authors records which are missing one?",
+            abort=True,
+        )
+    current_app.config["FEATURE_FLAG_ENABLE_BAI_CREATION"] = True
+    all_authors = PersistentIdentifier.query.filter_by(
+        pid_type="aut", status="R"
+    ).with_entities("object_uuid")
+    bai_authors = PersistentIdentifier.query.filter_by(
+        pid_type="bai", status="R"
+    ).with_entities("object_uuid")
+    all_authors = {author[0] for author in all_authors}
+    bai_authors = {author[0] for author in bai_authors}
+    missing_bais = all_authors - bai_authors
```

Comment:
this should be a DB join, similar to what we have in https://github.com/inspirehep/inspirehep/blob/7150712b6f2b4204d8e08da0fad406257901c8e7/backend/inspirehep/records/utils.py#L111-L120

Reply:
Discussed on zoom